tests/devices/unit_tests/test_eiger.py

```diff
@@ -544,14 +549,19 @@ def test_given_in_free_run_mode_and_not_all_frames_collected_in_time_when_unstag
 def test_if_arming_in_progress_then_stage_waits_for_completion(
     fake_eiger: EigerDetector, mock_set_odin_filewriter
 ):
-    def set_bad_arming_status():
-        fake_eiger.arming_status = get_bad_status()
+    fake_eiger.arming_status = Status()
 
-    assert fake_eiger.arming_status.done is True
-    fake_eiger.arming_status = get_bad_status()
-    # Should do .wait and error
-    with pytest.raises(Exception):
-        fake_eiger.stage()
+    thread = threading.Thread(target=fake_eiger.stage, daemon=True)
```

Comment:
could: would be nice to add a brief comment explaining what is happening in each thread.